libs/wingc/src/type_check.rs

```diff
@@ -4667,34 +4667,25 @@ new cloud.Function(@inflight("./handler.ts"), lifts: { bucket: ["put"] });
 				// Make sure there's a `super()` call to the parent ctor
 				if parent_ctor_sig.min_parameters() > 0 {
 					// Find the `super()` call
-					if let Some((idx, super_call)) = ctor_body
+					if let Some((idx, _super_call)) = ctor_body
 						.statements
 						.iter()
 						.enumerate()
 						.find(|(_, s)| matches!(s.kind, StmtKind::SuperConstructor { .. }))
 					{
-						// We have a super call, make sure it's the first statement (after any type defs)
-						let expected_idx = ctor_body
-							.statements
-							.iter()
-							.position(|s| !s.kind.is_type_def())
-							.expect("at least the super call stmt");
-						if idx != expected_idx {
-							self.spanned_error(
-								super_call,
-								format!(
-									"super() call must be the first statement of {}'s constructor",
-									ast_class.name
-								),
-							);
+						// Check if one of the statements before the super() call is invalid
+						for i in 0..idx {
+							if self.type_check_invalid_stmt_before_super(&ctor_body.statements[i]) {
+								self.spanned_error(
+									&ctor_body.statements[i],
+									"This command cannot be executed before calling super()".to_string(),
```

Comment:
I'm guessing you don't need `to_string` here:
```suggestion
									"This command cannot be executed before calling super()",
```